frontend/learning/components/ItemDisplay/WordDisplay/StrokeGif.js

```diff
@@ -0,0 +1,127 @@
+
+import HanziWriter from 'hanzi-writer';
+import React, {useEffect, useState} from 'react';
+import styled from 'styled-components';
+import '@learning.styles/ItemDisplay.css';
+import PropTypes from 'prop-types';
+
+const WordContainer = styled.div`
+    font-size: 3.75 em;
+    font-weight: 200;
+    text-align: center;
+    display: flex;
+    flex-direction: row;
+`;
+
+/**
+ * @param {String} props.item
+ * @param {Object} props
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export default function StrokeGif(props) {
+
+    /* Split input into array of characters. */
+    const items = props.item.split('');
+
+    /* Create target ID for each of the characters. */
+    const itemsTargetIDs = items.map((value, index) => `writer-target-${index}`);
+
+    // an array of 'writers'
+    let writers = null; 
+
+    const [gifState1, setGifState1] = useState(0);
+    const [gifState2, setGifState2] = useState(0);
+    /* 1 - Minor Problem: How to create an array of states?
+    const [state, setState] = itemsTargetIDs.map((value, index) => {
+        [state[index], setState[index]] = useState(0);
+    }); 
+    */
+    
+    /* 2 - Main Problem: after initial useEffect, 'writers' object falls back to 'null';
+            When tired to pauseAnimation (didUpdate), 'writers' cannot be referred to.
+            Tried passing in a second parameter to induce didUpdate, but triggered re-rendering.
+            Thought of cleanup within useEffect, but could not get it working.
+    */
```

Comment:
React calls this component function every time it needs to be rendered, and your useEffect depends on `[]` which means that it will only be run on the very first render (which is good for now but I would say have them depend on props so that words could be updated with next button). To have `writers` persist, make them into a state and call setState in your useEffect.